tt_metal/tools/tt_builder/builder.cpp

```diff
@@ -0,0 +1,75 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#include <iostream>
+#include <fstream>
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/common/utils.hpp"
+#include "tt_metal/jit_build/build.hpp"
+#include "tt_metal/impl/device/device.hpp"
+#include "tt_metal/tools/tt_builder/builder.hpp"
+
+using namespace tt;
+namespace fs = std::filesystem;
+using namespace std;
+
+namespace tt::tt_metal {
+tt_builder::tt_builder() : output_dir_("/tmp/tt-metal-cache/") {}
```

Comment:
I am currently using `/tmp/tt-metal-cache/` as the default cache path for binaries.
Plan is to change it to `built` once I add the git hash to this tool. Thoughts?
The path can also be changed by user-input.

Reply:
I think that makes sense.  So:
 - finish the other work to move built to /tmp (w/ an env var override)
 - put this build into the built dir.  should go under ".../built/offline/" w/ subdirs for firmware and kernels